app/src/main/java/ru/chaichuk/hwapp/viewModels/MoviesListViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package ru.chaichuk.hwapp.viewModels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.launch
+import ru.chaichuk.hwapp.data.Movie
+import ru.chaichuk.hwapp.data.MoviesListLoader
+
+class MoviesListViewModel : ViewModel(){
+
+    var moviesListLoader: MoviesListLoader? = null
```

Comment:
why does the adapter need an inside ViewModel?
something strange logic

Reply:
I had two options:
1) pass application or activity context to ViewModel, as loadMovies(context) needs it
2) pass external loader as adapter, to not pass context
As I heard in lecture, it is not advisable to pass context to VewModel (or use AndroidViewModel) for easier testing, so I decided to pass adapter. I've also tried to pass context, and tried variant with global static application context. But leaved final variant with adapter.
What is better variant to call loadMovies(context)?